deploy.js

```diff
@@ -7,16 +7,68 @@ const {
   executeCommands,
 } = require('@hollowverse/common/helpers/executeCommands');
 const { createZipFile } = require('@hollowverse/common/helpers/createZipFile');
+const fs = require('fs');
+const awsSdk = require('aws-sdk');
 
-const { IS_PULL_REQUEST } = shelljs.env;
+const {
+  IS_PULL_REQUEST,
+  AWS_REGION = 'us-east-1',
+  CLOUDFRONT_DISTRIBUTION_ID,
+} = shelljs.env;
 
 const isPullRequest = IS_PULL_REQUEST !== 'false';
 
 async function main() {
-  const buildCommands = ['yarn clean', 'yarn build'];
+  const buildCommands = ['yarn clean', 'yarn test'];
   const deploymentCommands = [
+    'yarn build',
     () => createZipFile('build.zip', ['dist/**/*'], ['secrets/**/*.enc']),
-    'aws lambda update-function-code --function-name assignEnvironment --zip-file fileb://build.zip --publish',
+    async () => {
+      const cloudfront = new awsSdk.CloudFront({
+        apiVersion: '2017-03-25',
+        region: AWS_REGION,
+      });
+      const lambda = new awsSdk.Lambda({
+        apiVersion: '2015-03-31',
+        region: AWS_REGION,
+      });
+
+      const { FunctionArn } = await lambda
+        .updateFunctionCode({
+          FunctionName: 'assignEnvironment',
+          Publish: true,
+          ZipFile: fs.readFileSync('build.zip'),
+        })
+        .promise();
+
+      const { DistributionConfig, ETag } = await cloudfront
+        .getDistributionConfig({
+          Id: CLOUDFRONT_DISTRIBUTION_ID,
+        })
+        .promise();
+
+      await cloudfront
+        .updateDistribution({
+          Id: CLOUDFRONT_DISTRIBUTION_ID,
+          IfMatch: ETag,
+          DistributionConfig: {
+            ...DistributionConfig,
+            DefaultCacheBehavior: {
+              ...DistributionConfig.DefaultCacheBehavior,
+              LambdaFunctionAssociations: {
+                Quantity: 1,
+                Items: [
+                  {
+                    EventType: 'viewer-request',
+                    LambdaFunctionARN: FunctionArn,
+                  },
+                ],
```

Comment:
I wonder if in the future we ever needed more Lambda function associations per CF distribution, will this overwrite any existing associations? If that's true, maybe it's better to `push` an association into the `LambdaFunctionAssociations` array, and increment the `Quantity` by one. Just a thought.

Reply:
It's not as simple as pushing a new item. We need to replace the previous version. We should look in `Items` for an association that matches `FunctionArn` with the version suffix removed.